rq_scheduler/scheduler.py

```diff
@@ -365,23 +365,45 @@ def enqueue_job(self, job):
             job.meta['repeat'] = int(repeat) - 1
 
         queue = self.get_queue_for_job(job)
-        queue.enqueue_job(job)
-        self.connection.zrem(self.scheduled_jobs_key, job.id)
 
-        if interval:
-            # If this is a repeat job and counter has reached 0, don't repeat
-            if repeat is not None:
-                if job.meta['repeat'] == 0:
-                    return
-            self.connection.zadd(self.scheduled_jobs_key,
-                                  {job.id: to_unix(datetime.utcnow()) + int(interval)})
-        elif cron_string:
-            # If this is a repeat job and counter has reached 0, don't repeat
-            if repeat is not None:
-                if job.meta['repeat'] == 0:
+        with self.connection.pipeline() as pipe:
+            while True:
+                try:
+                    pipe.watch(self.scheduled_jobs_key)
+                    queue.enqueue_job(job)
+                    pipe.multi()
+                    pipe.zrem(self.scheduled_jobs_key, job.id)
+
+                    if not (interval or cron_string):
+                        pipe.execute()
+                        return
+
+                    # If this is a repeat job and counter has reached 0, don't repeat
+                    if repeat is not None:
+                        if job.meta['repeat'] == 0:
+                            pipe.execute()
+                            return
+
+                    next_scheduled_time = (
```

Comment:
Could you explain the rationale behind changing the way which `next_scheduled_time` calculation is different than the current implementation?